### PR TITLE
User Story 18 

### PR DESCRIPTION
**Pull request by WillWeston94**
Some conflicts in here I need to solve

I will solve then post the branch once all necessary changes are configured

### CODE REVIEW

**Comment by WillWeston94** [comment]
Going to walk through the files changed now to look at all the new data

File: app/views/merchant_invoices/index.html.erb
Comment (WillWeston94) [inline app/views/merchant_invoices/index.html.erb:1]: This was done so we had unique invoices showing no repeaters on the index

File: app/views/merchant_invoices/show.html.erb
Comment (WillWeston94) [inline app/views/merchant_invoices/show.html.erb:1]: Total Revenue and Form

File: app/models/invoice_item.rb
Comment (WillWeston94) [inline app/models/invoice_item.rb:1]: Just swapped hash to array for index positon instead of case use on the rake

File: config/application.rb
Comment (WillWeston94) [inline config/application.rb:1]: Got rid of some CSS tailwind stuff that I will end up adding later for an extension

File: config/routes.rb
Comment (WillWeston94) [inline config/routes.rb:1]: patch route for update action

File: lib/tasks/invoice_items.rake
Comment (WillWeston94) [inline lib/tasks/invoice_items.rake:1]: Deleted case statement as I turned hash into array

File: spec/features/admin/merchants/show_spec.rb
Comment (WillWeston94) [inline spec/features/admin/merchants/show_spec.rb:1]: Replace original tests from last week with factory bot instances

File: spec/features/merchant_invoices/show_spec.rb
Comment (WillWeston94) [inline spec/features/merchant_invoices/show_spec.rb:1]: describe "displays a select field to update the item's status " do
      xit "and when I click that select field I can select a new status and click a button to update the item's status" do
      
      this test has Capybara CSS fail for button status

**Review by WillWeston94 — COMMENTED** [review]
See single line comments for specificities

**Comment by kicotiel** [comment]
Finalized the branch by handling some really weird errors with enums. I simplified the forms using them to update invoice and invoice_items statuses.

Also merged main in, took care of the doubled up ready_to_ship test and reorganized the merchant model spec file to have all tests in the correct categories.

**Pull request merged**